onnx/onnxifi_ext.h

```diff
@@ -0,0 +1,163 @@
+#ifndef ONNXIFI_EXT_H
+#define ONNXIFI_EXT_H 1
+
+#include "onnx/onnxifi.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Generic ONNXIFI extension function pointer.
+ *
+ * The caller should convert this generic function pointer to the function
+ * pointer specific for an extension function type.
+ */
+typedef onnxStatus (ONNXIFI_ABI* onnxExtensionFunctionPointer)(void);
+
+/**
+ * A combination of onnxSetIO and onnxRunGraph, functionally equals to first run
+ * onnxSetIO(graph, inputsCount, inputDescriptors, outputsCount, outputDescriptors),
+ * then run onnxRunGraph(graph, inputFence, outputFence)
+ * @param graph - graph handle created by onnxInitGraph.
+ * @param inputsCount - number of elements in the inputDescriptors array.
+ * @param[in] inputDescriptors - descriptors of input tensors for the graph.
+ *                               Elements of this array must provide a location
+ *                               for each ValueInfoProto.name listed in
+ *                               ModelProto.graph.input of the ONNX graph.
+ *                               If inputsCount is non-zero, inputDescriptors
+ *                               pointer must be non-NULL.
+ * @param outputsCount - number of elements in the outputDescriptors array.
+ *                       Must be greater than zero.
+ * @param[in] outputDescriptors - descriptors of output tensors for the graph.
+ *                                outputDescriptors pointer must be non-NULL.
+ *                                Elements of this array must provide a location
+ *                                for each ValueInfoProto.name listed in
+ *                                ModelProto.graph.output of the ONNX graph.
+ * @param[in] inputFence - synchronization primitive that signals when graph
```

Comment:
do we need to care about this? or should inputs be available right away on the `onnxSetIOAndRunGraph` call

Reply:
Yes it would be cool if we could get rid of this also if possible

Reply:
or if we have that param for general purpose in this API, we can make C2/Glow to make sure that Glow does not need to wait on that event.
That option also works well.

Reply:
Quite a good point. My thought is since this is a general API, we should better give less restrictions or potential bug maker to our backend, so I would like to remove this InputFence from this API for now. If adding this param become very important in the future, I guess we can add it back then. Agree?

Reply:
That sounds good to me.